tests/conftest.py

```diff
@@ -2981,6 +2981,7 @@ def finalizer():
     request.addfinalizer(finalizer)
 
 
+@retry.retry(KeyError)
```

Comment:
This is bad, a retry already happens from the calling code in `nb_ensure_endpoint_count`, we just need to not throw and return False

Reply:
good point. Fixed now